File: koola10/stellar/balance.go
package stellar

import (
	"github.com/stellar/go/clients/horizonclient"
	"github.com/stellar/go/protocols/horizon"
)

func CheckBalance(address string) ([]horizon.Balance, error) {
	client := GetClient()
	request := horizonclient.AccountRequest{AccountID: address}
	account, err := client.AccountDetail(request)
	if err != nil {
		return nil, err
	}
	return account.Balances, nil
}

File: backend/gateway/src/stellar/balance.ts
import { server } from './client.js';

export async function checkBalance(publicKey: string): Promise<any[]> {
    try {
        const account = await server.loadAccount(publicKey);
        return account.balances;
    } catch (e) {
        console.error('Error loading account:', e);
        throw e;
    }
}
